Irenajo/enemy

irenajo opened the pull request:
Added a TargetTile, if enemies reach it they despawn and deal damage to Player. 
Added collider which will interract with turrets/bullets.
Added takeDamage, dealDamage, onTriggerEnter functions, Destroyed/atTarget states.
The EnemyManager now updates all enemies.
Added code comments.

irenajo committed:
added target, enemyManager control, enemy collider

irenajo committed:
Added code comments

Gojicluka commented on Assets/Scripts/EnemyManager.cs:
```diff
@@ -1,7 +1,38 @@
+
+    void Update()
+    {
+        for (int i = enemies.Count - 1; i >= 0; i--)
```
moving the enemies should be done from the enemy class individually, since there is no reason for it to be done in enemy manager. This would make sense if enemy classes didn't have access to the update method, but since they do, splitting the functionality like this would lead to harder readability.

irenajo committed:
Added simple UI

irenajo committed:
Removed enemyManager and implemented UI wave start

irenajo commented:
- Added simple UI that can start a wave on button click.
- Moved Enemy logic from EnemyManager to Enemy script.
- Removed EnemyManager.
- Implemented GameManager that manages spawners and player.
